File: src/Neuralm.Presentation.Web/src/messages/requests/Request.ts
import Guid from '@/helpers/Guid';
import { IMessage } from '@/interfaces/IMessage';

/**
 * Represents the Request class.
 */
export default class Request implements IMessage {
  public id: string;
  public dateTime: Date;

  /**
   *  Initializes the Request class.
   */
  constructor() {
    this.id = Guid.newGuid().toString();
    this.dateTime = new Date();
  }
}

File: src/Neuralm.Services/Neuralm.Services.Common.Messages/Abstractions/Request.cs
﻿using Neuralm.Services.Common.Messages.Interfaces;
using System;

namespace Neuralm.Services.Common.Messages.Abstractions
{
    /// <summary>
    /// Represents the <see cref="Request"/> class.
    /// </summary>
    public abstract class Request : IMessage
    {
        /// <summary>
        /// Gets the id.
        /// </summary>
        public Guid Id { get; set; }

        /// <summary>
        /// Gets the date time.
        /// </summary>
        public DateTime DateTime { get; set; }

        /// <summary>
        /// Initializes an instance of the <see cref="Request"/> class.
        /// </summary>
        protected Request()
        {
            Id = Guid.NewGuid();
            DateTime = DateTime.UtcNow;
        }
    }
}
